src/transformers/modeling_blenderbot.py

```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/env python3
+# coding=utf-8
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the;
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# LICENSE file in the root directory of this source tree.
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from .configuration_blenderbot import BlenderbotConfig
+from .file_utils import add_start_docstrings_to_callable
+from .modeling_bart import (
+    BartDecoder,
+    BartEncoder,
+    SinusoidalPositionalEmbedding,
+    _prepare_bart_decoder_inputs,
+    _reorder_buffer,
+)
+from .modeling_outputs import BaseModelOutput, Seq2SeqLMOutput
+from .modeling_utils import PreTrainedModel
+
+
+class PretrainedBlenderbotModel(PreTrainedModel):
+    config_class = BlenderbotConfig
+    base_model_prefix = "."
+
+    def _init_weights(self, module):
+        std = self.config.init_std
+        if isinstance(module, (nn.Linear, nn.Embedding)):
+            module.weight.data.normal_(mean=0.0, std=std)
+        elif isinstance(module, SinusoidalPositionalEmbedding):
+            pass
+        if isinstance(module, nn.Linear) and module.bias is not None:
+            module.bias.data.zero_()
+        elif isinstance(module, nn.Embedding) and module.padding_idx is not None:
+            module.weight.data[module.padding_idx].zero_()
+
+
+class BlenderbotOutput(nn.Module):
+    def __init__(self, config: BlenderbotConfig, embeddings):
+        super().__init__()
+        self.embeddings = embeddings
+        self.bos_token_id = config.bos_token_id
+
+    def forward(self, hidden_states):
+        output = F.linear(hidden_states, self.embeddings.weight)
+        # we need to force their probability of generation to be 0.
+        output[:, :, self.bos_token_id] = -65504 if output.dtype is torch.float16 else -1e20
```

Comment:
better to work with `torch.finfo(torch.float16).max` here instead of hardcoding parameters